ProcessOptimizer/space/constraints.py

```diff
@@ -476,6 +630,55 @@ def __eq__(self, other):
         else:
             return False
 
+class SumEquals():
+    def __init__(self, dimensions, value):
```

Comment:
Have we decided not to use type hints when we add new functions? If so, I think that is a mistake, but I will bow to the decision.

Reply:
No, I had just forgotten about it again. It may take some time for me to get used to how to correctly use this concept for inputs that aren't very explicitly given as examples in PEP-484, so bear with me when I make mistakes in future pull requests and/or forget to use it.

One thing that will inevitably happen is that the first line of function definitions will no longer fit on a single line, so correct formatting there will probably also be a bit all over the place for a while until I nail that.

Reply:
Great!

That is perfectly fine. I apologise if I seemed a bit gruff in my earlier answer, I have been unreasonably grumpy today, and I shouldn't have let that shine through. I am sorry.

Non-optimal type are fine, it gets very complicated fast. I am also all over the place when I have to decide what abstract class an input should be. I usually start out very concrete (this is a list of integers), and only later remember that perhaps a Sequence is enough. Or even an Iterable? What was the difference between those two again? And then I am off trying to google that, and interpret what that difference means for my concrete code.

Yes, it easily makes the input argument longer than the standard line length, which is annoying.